ios/FluentUI/Command Bar/CommandBar.swift

```diff
@@ -0,0 +1,290 @@
+        layer.locations = [0, 0, 1] // Initially the entire layer is white color, so the view is not masked.
+
+        return layer
+    }()
+}
+
+// MARK: - Scroll view delegate
+
+extension CommandBar: UIScrollViewDelegate {
+    public func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        updateShadow()
+    }
+}
+
+// MARK: - Private members
+
+private extension CommandBar {
+    func configureHierarchy() {
+        addSubview(containerView)
+
+        // Left and right button layout constrants
+        if let leadingButton = leadingButton {
+            addSubview(leadingButton)
+            NSLayoutConstraint.activate([
+                leadingButton.topAnchor.constraint(equalTo: containerView.topAnchor),
+                leadingButton.leadingAnchor.constraint(equalTo: leadingAnchor, constant: CommandBar.fixedButtonSpacing),
+                containerView.bottomAnchor.constraint(equalTo: leadingButton.bottomAnchor)
+            ])
+        }
+
+        if let trailingButton = trailingButton {
+            addSubview(trailingButton)
+            NSLayoutConstraint.activate([
+                trailingButton.topAnchor.constraint(equalTo: containerView.topAnchor),
+                containerView.bottomAnchor.constraint(equalTo: trailingButton.bottomAnchor),
+                trailingAnchor.constraint(equalTo: trailingButton.trailingAnchor, constant: CommandBar.fixedButtonSpacing)
+            ])
+        }
+
+        // Button container layout constrants
+        let containerLeadingConstraint: NSLayoutConstraint = {
+            if let leadingButton = leadingButton {
+                return containerView.leadingAnchor.constraint(equalTo: leadingButton.trailingAnchor, constant: CommandBar.fixedButtonSpacing)
+            } else {
+                return containerView.leadingAnchor.constraint(equalTo: leadingAnchor)
+            }
+        }()
+
+        let containerTrailingConstraint: NSLayoutConstraint = {
+            if let trailingButton = trailingButton {
+                return trailingButton.leadingAnchor.constraint(equalTo: containerView.trailingAnchor, constant: CommandBar.fixedButtonSpacing)
+            } else {
+                return trailingAnchor.constraint(equalTo: containerView.trailingAnchor)
+            }
+        }()
+
+        NSLayoutConstraint.activate([
+            containerView.topAnchor.constraint(equalTo: topAnchor, constant: CommandBar.insets.top),
+            bottomAnchor.constraint(equalTo: containerView.bottomAnchor, constant: CommandBar.insets.bottom),
+
+            containerLeadingConstraint,
+            containerTrailingConstraint
+        ])
+
+        stackView.layoutIfNeeded()
+
+        if UIView.userInterfaceLayoutDirection(for: semanticContentAttribute) == .rightToLeft {
+            // Flip the scroll view to invert scrolling direction. Flip its content back because it's already in RTL.
+            let flipTransform = CGAffineTransform(scaleX: -1, y: 1)
```

Comment:
it RTL layout does the scroll direction change and the ordering of the items itself change?

Reply:
RTL on iOS doesn't change the scroll direction but `UIStackView` would change the ordering of its items.